### PR TITLE
megamenu fixes

### PR DESCRIPTION
**Pull request by vineethvanga18**
Signed-off-by: vineethvanga18 <reddy.8@iitj.ac.in>

**Description**

- fetching of blogs dynamically

- fixes and improvements in animations

- minor naming and UI fixes

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by shivaylamba — COMMENTED** [review]
image
@vineethvanga18 the images are a bit stretched. I think we can make the mega menu slightly smaller as well.

**Comment by vineethvanga18** [comment]
yes @shivaylamba, One solution would be to fetch the correct sized image through graphql

**Comment by leecalcote** [comment]
Thank you, @vineethvanga18. Two other issues to address:

1. When the mouse approaches the menu from underneath, the menu is activated well in advance of the mouse reaching the menu text.
1. The exit animation is not invoked when the mouse leaves from the top of the menu.

Dec-04-2020 17-56-30

**Comment by kishore007k** [comment]
Nice Work @Vineet-Sharma29 
One small change, It would be nice if we display the upcoming events/workshops in the community instead of displaying the most recent ones.
What are your thoughts @leecalcote Sir?

**Comment by Nikhil-Ladha** [comment]
> Nice Work @Vineet-Sharma29
> One small change, It would be nice if we display the upcoming events/workshops in the community instead of displaying the most recent ones.
> What are your thoughts @leecalcote Sir?

+1. Good point @kishore007k

File: src/sections/Navigation/index.js [hunk 1]
```diff
@@ -11,6 +12,102 @@ import ScrollspyMenu from "./utility/ScrollspyMenu.js";
...
+                    sort: { fields: [frontmatter___date], order: DESC }
+                    filter: { fields: { collection: { eq: "projects" } } }
+                    limit: 2
+                ) {
+                    nodes {
+                        id
+                        frontmatter {
+                            title
+                            thumbnail{
+                                publicURL
```
Comment (Tanuj22) [inline src/sections/Navigation/index.js:28]: Here childImageSharp should be used like (reference can be taken from other queries)

File: src/sections/Navigation/utility/ScrollspyMenu.js [hunk 2]
```diff
@@ -51,12 +65,12 @@ const ScrollspyMenu = ({ menuItems, ...props }) => {
                             </div>
                             <div className="nav-display">
                                 <div className="nav-div">
-                                    <a href={active.div1.path}><img src={active.div1.src} className="nav-img" /></a>
-                                    <div> {active.div1.descr} </div>
+                                    <Link href={blogData[active.name].nodes[0].fields.slug}><img src={blogData[active.name].nodes[0].frontmatter.thumbnail.publicURL} className="nav-img" /></Link>
```
Comment (Tanuj22) [inline src/sections/Navigation/utility/ScrollspyMenu.js:68]: After using childImageSharp. `Image` component can be used

Comment (vineethvanga18) [reply src/sections/Navigation/utility/ScrollspyMenu.js:68]: What are the available image sizes @Tanuj22

Comment (Tanuj22) [reply src/sections/Navigation/utility/ScrollspyMenu.js:68]: Any size will work

**Review by leecalcote — APPROVED** [review]
We do need to reduce the overall size of this menu, but in its current form it is pretty **_awesome_**, @vineethvanga18.  🥇

**Pull request merged**